### PR TITLE
python3Packages.pywlroots: 0.16.8 -> 0.17.0, python3Packages.qtile: 0.25.0 -> 0.26.0, python3Packages.qtile-extras: 0.25.0 -> 0.26.0

### PR DESCRIPTION
**Pull request by arjan-s**
## Description of changes
Update `pywlroots`, `qtile` and `qtile-extras` in tandem. These releases are tightly coupled, hence the combined PR.

https://github.com/flacjacket/pywlroots/releases/tag/v0.17.0
https://github.com/qtile/qtile/releases/tag/v0.26.0
https://github.com/elParaguayo/qtile-extras/releases/tag/v0.26.0

Everything builds fine, I'll test using the new version tonight.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -12911,9 +12909,7 @@ self: super: with self; {
 
   qtconsole = callPackage ../development/python-modules/qtconsole { };
 
-  qtile = callPackage ../development/python-modules/qtile {
-    wlroots = pkgs.wlroots_0_16;
-  };
+  qtile = callPackage ../development/python-modules/qtile { };
```
Comment (jwijenbergh) [inline pkgs/top-level/python-packages.nix:12912]: Pass wlroots_0_17 instead. Sway also does this

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -12785,9 +12785,7 @@ self: super: with self; {
 
   pywizlight = callPackage ../development/python-modules/pywizlight { };
 
-  pywlroots = callPackage ../development/python-modules/pywlroots {
-    wlroots = pkgs.wlroots_0_16;
-  };
+  pywlroots = callPackage ../development/python-modules/pywlroots { };
```
Comment (jwijenbergh) [inline pkgs/top-level/python-packages.nix:12788]: Pass wlroots_0_17

**Comment by jwijenbergh** [comment]
Thanks Arjan? Could you add libcst to the python dependencies for qtile? It's needed for `qtile migrate`

**Comment by arjan-s** [comment]
> Thanks Arjan? Could you add libcst to the python dependencies for qtile? It's needed for `qtile migrate`

Added now, but `qtile migrate` doesn't work yet: `ModuleNotFoundError: No module named 'libcst.native'`
Will look into this later.

File: pkgs/development/python-modules/qtile/default.nix [hunk 3]
```diff
@@ -70,6 +71,7 @@ buildPythonPackage rec {
     dbus-next
     dbus-python
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/qtile/default.nix:72]: I think dbus-python is unused and can be removed.

File: pkgs/development/python-modules/qtile/default.nix [hunk 4]
```diff
@@ -70,6 +71,7 @@ buildPythonPackage rec {
...
     iwlib
+    libcst
     mpd2
     psutil
     pulsectl-asyncio
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/qtile/default.nix:77]: I think based on the code pytz is preferred over dateutil https://github.com/qtile/qtile/blob/e4b4bb8b6f2c07570b049e7e9171f5ce6da0625c/libqtile/widget/clock.py#L39

File: pkgs/development/python-modules/qtile/default.nix [hunk 5]
```diff
@@ -70,6 +71,7 @@ buildPythonPackage rec {
     dbus-next
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/qtile/default.nix:71]: Can we replace the --replace with --replace-fail?

**Review by SuperSandro2000 — COMMENTED** [review]
I've just looked through the repo, so my assumptions might be wrong.

File: pkgs/development/python-modules/qtile/default.nix (hunk 3 above)
Comment (jwijenbergh) [reply pkgs/development/python-modules/qtile/default.nix:72]: One of the qile maintainers here, correct. Good catch

**Comment by arjan-s** [comment]
Comments addressed. But didn't have time to actually test run the new version yet :(

**Comment by Sigmanificient** [comment]
@ofborg build qtile-unwrapped.passthru.tests

**Comment by Sigmanificient** [comment]
image

**Comment by arjan-s** [comment]
Also, added `python-dateutil` to `qtile-extras` deps, because [one of its tests depends on it](https://github.com/elParaguayo/qtile-extras/blob/1df9f1b017cd453254326acb948df4d0fc2f8965/test/widget/test_livefootballscores.py#L380)

**Comment by Sigmanificient** [comment]
@ofborg build qtile-unwrapped.passthru.tests

**Comment by Sigmanificient** [comment]
(passthru have be be manually run because they are bound to `qtile-unwrapped`)

**Comment by arjan-s** [comment]
Update: I've tested these changes and they work perfectly.
This is ready to merge as far as I'm concerned.

**Comment by Sigmanificient** [comment]
The nixos-vm test should be changed to only work on linux platforms, will do in another pr

**Review by Sigmanificient — APPROVED** [review]

**Pull request merged**